extensions/src/main/java/org/pentaho/platform/plugin/services/pluginmgr/PentahoSystemPluginManager.java

```diff
@@ -365,12 +369,24 @@ private void registerPlugin( final IPlatformPlugin plugin ) throws PlatformPlugi
 
     ClassLoader loader = PentahoSystem.get( ClassLoader.class, null,
       Collections.singletonMap( PLUGIN_ID, plugin.getId() ) );
-    GenericApplicationContext beanFactory = PentahoSystem
-      .get( GenericApplicationContext.class, null, Collections.singletonMap( PLUGIN_ID, plugin.getId() ) );
+
+    Objects.requireNonNull( loader );
+
+    var baseName = plugin.getResourceBundleClassName();
+    // set the provider that will be used to load the plugin's resource bundles for localization
+    if ( !StringUtil.isEmpty( baseName ) ) {
+      plugin.setResourceBundleProvider( new CachingResourceBundleProvider(
+        ( Locale locale ) -> ResourceBundle.getBundle( baseName, Objects.requireNonNull( locale ), loader )
+      ) );
+    }
+
     createAndRegisterLifecycleListeners( plugin, loader );
 
     plugin.init();
 
+    GenericApplicationContext beanFactory = PentahoSystem
```

Comment:
This is now out of order. Previously, it was before `createAndRegisterLifecycleListeners`. Any reason?

Reply:
was moving closer to were it is used, will be moved back after rebasing with your recent assertion code